src/ucs/config/parser.c

```diff
@@ -1427,6 +1434,13 @@ ucs_status_t ucs_config_parser_fill_opts(void *opts, ucs_config_field_t *fields,
         goto err;
     }
 
+    UCS_INIT_ONCE(&config_file_parse) {
+        status = ucs_config_parse_config_file(UCX_CONF_FILE, 0);
+    }
+    if (status != UCS_OK) {
+        goto err;
+    }
```

Comment:
looks weird because status could be uninitialized here
maybe just ignore parse error?

Reply:
Ignoring a parse error is an option, I don't have any strong objection against (maybe at least issue a "warning"? Provide feedback to the user, for the case when he tweaked something in `ucx.conf` but the settings didn't apply because of conf file syntax error for example).

But - how can `status` be uninitialized here? There's a call to `ucs_config_parser_get_sub_prefix` above. The only problem with this code is checking `if (status != UCS_OK) {` twice which is not needed after the first initialization is done.

Reply:
1. i think better add a warning, since the 2nd time this function is used, the parse error is de-facto ignored
2. it's not really uninitialized - it's just the status could sometimes be set by previous code block, which is not a straightforward behavior, and better to avoid it altogether

Reply:
Done